### ISSUE TITLE
`Dense.Apply` doesn't work when length is 1

### ISSUE THREAD

**Issue by chewxy** [chewxy/gorgonia#144]
No idea why, but this was found from quickcheck testing (see `dense_apply_test.go`)

**Comment by chewxy** [comment]
This bug was due to a simple error in `genlib2`:


Original:
```
case as && !incr && f0 != nil:
                       at[0], err = f0(at[0])
```

Correct:
```
case as && !incr && f0 != nil:
                       at[0], err = f1(at[0])
```

**Comment by chewxy** [comment]
Fixed in bf089b5
